java-checks/src/main/java/org/sonar/java/checks/security/ServerCertificatesCheck.java

```diff
@@ -38,12 +38,22 @@
 @DeprecatedRuleKey(ruleKey = "S4424", repositoryKey = "java")
 @Rule(key = "S4830")
 public class ServerCertificatesCheck extends IssuableSubscriptionVisitor {
+  private static final String JAVA_LANG_STRING = "java.lang.String";
+  private static final String X509_CERTIFICATE_ARRAY = "java.security.cert.X509Certificate[]";
 
-  private static final MethodMatchers CHECK_MATCHER = MethodMatchers.create()
-    .ofSubTypes("javax.net.ssl.X509TrustManager")
-    .names("checkClientTrusted", "checkServerTrusted")
-    .addParametersMatcher("java.security.cert.X509Certificate[]", "java.lang.String")
-    .build();
+  private static final MethodMatchers TRUST_MANAGER_MATCHER = MethodMatchers.or(
+    MethodMatchers.create()
+      .ofSubTypes("javax.net.ssl.X509TrustManager")
+      .names("checkClientTrusted", "checkServerTrusted")
+      .addParametersMatcher(X509_CERTIFICATE_ARRAY, JAVA_LANG_STRING)
+      .build(),
+    MethodMatchers.create()
+      .ofSubTypes("javax.net.ssl.X509ExtendedTrustManager")
+      .names("checkClientTrusted", "checkServerTrusted")
+      .addParametersMatcher(X509_CERTIFICATE_ARRAY, JAVA_LANG_STRING, "java.net.Socket")
+      .addParametersMatcher(X509_CERTIFICATE_ARRAY, JAVA_LANG_STRING, "javax.net.ssl.SSLEngine")
+      .build()
+  );
 
   @Override
   public List<Tree.Kind> nodesToVisit() {
```

Comment:
I suppose you haven't face it yet. But each time there is a call to ```hasSemantic()``` we're removing this condition. It was useful before frontend was migrated to Eclipse now it's not actually emulating real behaviour, because Eclipse try to resolve as much as possible and put ```unknown``` in case uf incomplete semantic.
So now we're removing it and adding tests ```.withoutSemantic()``` (If it fails you need to review issues and understand why this is happening)